relay-server/src/services/buffer/mod.rs

```diff
@@ -203,26 +208,34 @@ impl EnvelopeBufferService {
     }
 
     /// Tries to pop an envelope for a ready project.
-    async fn try_pop(
-        &mut self,
+    async fn try_pop<'a>(
+        config: &Config,
         buffer: &mut PolymorphicEnvelopeBuffer,
-    ) -> Result<(), EnvelopeBufferError> {
+        services: Services,
```

Comment:
Could this work with `services: &Services`? Or would that cause borrowing problems?